src/main/java/hudson/plugins/ec2/EC2ComputerLauncher.java

```diff
@@ -75,4 +89,95 @@ public void launch(SlaveComputer slaveComputer, TaskListener listener) {
     protected abstract void launchScript(EC2Computer computer, TaskListener listener)
             throws AmazonClientException, IOException, InterruptedException;
 
+
+    /**
+     * This method is called after a node disconnects. See {@link ComputerLauncher#afterDisconnect(SlaveComputer, TaskListener)}
+     * This method is overriden to perform a check to see if the node that is disconnected is a spot instance and
+     * whether the disconnection is a spot interruption event. If it is a spot interruption event, the tasks that the
+     * node was processing will be resubmitted if a user selects the option to do so.
+     * @param computer
+     * @param listener
+     */
+    @Override
+    public void afterDisconnect(SlaveComputer computer, TaskListener listener) {
+        if (computer == null) return;  // potential edge case where computer is null
+
+        Slave node = computer.getNode();
+        if (node instanceof EC2SpotSlave) {
+
+            // checking if its an unexpected disconnection
+            final boolean isUnexpectedDisconnection = computer.isOffline() && computer.getOfflineCause()
+                    instanceof OfflineCause.ChannelTermination;
```

Comment:
Some of the customers complained that ``OfflineCause.ChannelTermination`` was not always triggered for spot interruption. You may be able to dig this further here: https://github.com/jenkinsci/ec2-fleet-plugin/issues/121